src/matgl/models/_m3gnet.py

```diff
@@ -215,16 +215,19 @@ def forward(
         g: dgl.DGLGraph,
         state_attr: torch.Tensor | None = None,
         l_g: dgl.DGLGraph | None = None,
+        return_all_layer_output: bool = False,
```

Comment:
Why is this needed? There is no compute cost to returning all layers right? Why not return all and then use the relevant layer?

Reply:
If we return all layer outputs, the default behavior of forward method will be changed.